app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    User.find(author_id).update(post_counter: Post.where(author_id).count)
+  end
```

Comment:
- [ ] Great job for making use of one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), `after_save`. it is also good to make this method `private`. Kindly make it private